### PR TITLE
Ej

### LINKED ISSUE #14: Charts should sort x axis by y value
Currently, the app uses a built-in package to display all charts. However, for some ungodly reason, the package doesn't allow sorting. The result is this:
image
We will probably have to find another package and "rebuild" the charts.

### LINKED ISSUE #25: All graphs should have titles

### LINKED ISSUE #30: Understanding the EJ map
How can we make the map and graph clearer here? These are census blocks on the graph & map, right? Why don't they match each other? Why are there so few BLCKGRPCE designations when you click around?

image

### LINKED ISSUE #46: Check that table headers are reader-friendly
They don't have to be perfect, but we should look at all of them and see what can be improved for easier understanding
image

### LINKED ISSUE #47: Change EJ percentages to %
image
Seems hard and low priority but would improve legibility. Wonder if we might do it with a regex for "PCT" ending?

### LINKED ISSUE #48: Consider using map only for EJ indices
The bar graph may not be usable, since the names of census blocks will not be meaningful to users. I think the only utility the bar graph serves is to show the distribution, but I'm not sure how valuable that is/if it might be just as easy to discern on the map.
image

### LINKED ISSUE #49: Remove "DISPEO" from EJ indicators
It's an intermediate variable used by EPA (so not useful) and will error

### LINKED ISSUE #50: Consider breaking apart or removing environmental indicators from socioeconomic indicators
I find it super confusing that the environmental and the socioeconomic indicators are lumped together generally. Maybe we should focus just on the socioeconomic and filter out the environmental ones for greater clarity.

I do have technical definitions for e.g. Ozone as well, though, if we did want to keep them: "Annual average of top ten maximum daily 8-hour ozone air concentrations in parts per billion" e.g. — I guess the point is to identify facilities in areas where these indicators are already showing issues? If we want to keep, maybe that would be a good additional section so we might give more context on it?

_Originally posted by @Frijol in https://github.com/ericnost/streamlit_test/issues/29#issuecomment-1633549954_

### LINKED ISSUE #51: Remove "Most Frequently Reported Pollutants in Watershed" from Watershed Page?
Or move it to the bottom of the page? It doesn't line up well with the map anymore. The map now shows just the facilities reporting the selected pollutant, which is selected from the third component on the page. This should probably be moved up to the second component, just below the map, so the connection between the two (selecting a different pollutant produces a different map of facilities releasing that pollutant) is clear. The overall context - the number of facilities reporting each pollutant - could go at the bottom.

### PR DESCRIPTION
**Pull request by ericnost**
Several edits to streamline the tables and charts and improve their legibility, especially on the EJ page

Fixes #50
Fixes #49 
Fixes #48 
Fixes #47 
Fixes #46 
Fixes #30 
Fixes #25
Fixes #14
Fixes #51 in that I think we should just move that section (Most frequently reported pollutants) to the bottom of the apge

### CODE REVIEW

**Comment by ericnost** [comment]
Also fixes #51 in that I think we should just move that section (Most frequently reported pollutants) to the bottom of the apge

**Comment by Frijol** [comment]
This looks good! I think we do still want to keep the names of the variables (e.g. % low income) for use on the map and in the select box, rather than replacing everything with the full definition: 
image

File: pages/3_⚖️_Environmental_Justice.py [hunk 1]
```diff
@@ -72,21 +70,75 @@ def add_spatial_data(url, name, projection=4326):
...
-  census_data.set_index("GEOID", inplace=True)
-  ej_data.set_index("ID", inplace=True)
-  census_data = census_data.join(ej_data)
+  ej_data["ID"] = ej_data["ID"].astype(str) # set the Census id as a string
+  census_data.set_index("GEOID", inplace=True) # set it to the index in the Census data
+  ej_data.set_index("ID", inplace=True) # set the Census id to the index in the EJScreen data
+  census_data = census_data.join(ej_data) # join based on this shared id
+  census_data.rename(columns=ejdefs, inplace=True) # rename columns to the EJ variable definitions
+  census_data = census_data[[i for i in ejdefs.values()] + ["geometry"]] # Filter out unnecessary columns
+  census_data[[i for i in ejdefs.values()]] = round(census_data[[i for i in ejdefs.values()]] * 100, 2) # Convert decimal values to percentages
```
Comment (Frijol) [inline pages/3_⚖️_Environmental_Justice.py:133]: Also need to append "%" before displaying in the map

Comment (ericnost) [reply pages/3_⚖️_Environmental_Justice.py]: Done

File: pages/3_⚖️_Environmental_Justice.py [hunk 2]
```diff
@@ -114,95 +142,35 @@ def main():
...
-    @st.cache_data
-    def get_metadata():
-      columns = pd.read_csv("https://raw.githubusercontent.com/edgi-govdata-archiving/ECHO-SDWA/main/2021_EJSCREEEN_columns-explained.csv")
-      return columns
-    columns = get_metadata()
-    columns = columns.loc[columns["GDB Fieldname"].isin(ej_parameters)][["GDB Fieldname", "Description"]]
-    columns.set_index("Description", inplace = True)
-    ej_dict = columns.to_dict()['GDB Fieldname']
-
-    options = ej_dict.keys() # list of EJScreen variables that will be selected
```
Comment (Frijol) [inline pages/3_⚖️_Environmental_Justice.py:151]: Wait, we still want this dictionary as well! "ejvar" is the behind-the-scenes variable, "ej_dict" is the human-readable name for the category, and "ejdefs" is the longform description

**Pull request merged**